workers/social/lib/social/models/computeproviders/stacktemplate.coffee

```diff
@@ -400,6 +402,55 @@ module.exports = class JStackTemplate extends Module
       else callback null, provider.template
 
 
+  # returns preview stack template for given template and custom variables
+  #
+  # @param {Object} options
+  #   options for generating template preview
+  #
+  # @option options [String] template content
+  # @option options [Object] custom variables
+  #
+  # @return {Object} stacktemplate preview in provided format
+  #
+  # @example api
+  #
+  #   {
+  #     "template": "Hello ${var.koding_user_username} -- ${var.custom_foo}",
+  #     "custom": {"foo": "bar"}
+  #   }
+  #
+  # @example return
+  #
+  #   {
+  #     "template": "Hello gokmen -- bar",
+  #     "errors": {},
+  #     "warnings": {}
+  #   }
+  #
```

Comment:
block comment would be better for usage description, unless we do this on other places.